stock_cancel/stock.py

```diff
@@ -49,6 +49,10 @@ def action_revert_done(self):
                     _('Picking %s has invoices!') % (picking.name))
             picking.move_lines.write({'state': 'draft'})
             picking.state = 'draft'
+            for move in picking.move_lines:
+                for quant in picking.move_lines.quant_ids:
+                    if move.location_dest_id.id == quant.location_id.id:
+                        quant.location_id = move.location_id.id
             if picking.invoice_state == 'invoiced' and not picking.invoice_id:
                 picking.invoice_state = '2binvoiced'
             # Deleting the existing instance of workflow
```

Comment:
There is no more workflow on stock.picking in v8, so I guess we can delete the 2 lignes trg_delete and trg_create.